docs/decisions/0045-updated-vector-store-design.md

```diff
@@ -0,0 +1,810 @@
+class WeaviateMemoryCollectionService: IMemoryCollectionService;
+
+// Customers can inherit from our implementations and replace just the creation scenarios to match their schemas.
+class CustomerCollectionsService: AzureAISearchMemoryCollectionService, IMemoryCollectionService;
+
+// We can also create implementations that create indices based on an MLIndex specification.
+class MLIndexAzureAISearchMemoryCollectionService(MLIndex mlIndexSpec): AzureAISearchMemoryCollectionService, IMemoryCollectionService;
+
+interface IMemoryRecordService<TDataModel>
+{
+    Task<TDataModel?> GetAsync(string key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    Task DeleteAsync(string key, DeleteRecordOptions? options = default, CancellationToken cancellationToken = default);
+    Task<string> UpsertAsync(TDataModel record, UpsertRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchMemoryRecordService<TDataModel>(): IMemoryRecordService<TDataModel>;
+```
+
+#### Option 3 - Separated collection and record management with collection create separate from other operations.
+
+Vector store same as option 2 so not repeated for brevity.
+
+```cs
+
+interface IMemoryCollectionCreateService
+{
+    virtual Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchChatHistoryCollectionCreateService: IMemoryCollectionCreateService;
+class AzureAISearchSemanticCacheCollectionCreateService: IMemoryCollectionCreateService;
+
+// Customers can create their own creation scenarios to match their schemas, but can continue to use our get, does exist and delete class.
+class CustomerChatHistoryCollectionCreateService: IMemoryCollectionCreateService;
+
+interface IMemoryCollectionUpdateService
+{
+    IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default);
+    Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default);
+    Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class RedisMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class WeaviateMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+
+```
+
+#### Option 4 - Separated collection and record management with collection create separate from other operations, with collection management aggregation class on top.
+
+Variation on option 3. 
+
+```cs
+
+interface IMemoryCollectionCreateService
+{
+    virtual Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+interface IMemoryCollectionUpdateService
+{
+    IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default);
+    Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default);
```

Comment:
Why not  `GetCollectionAsync` and return some metadata?

Reply:
Typically, the only thing valuable here to return is the index configuration.  That varies significantly between different dbs, so having a common description would mean we would lose data as we convert.  Losing data is bad if we also support create, since a developer could rightly assume that they should be able to get, modify and create, but they would get something that deviates from the original in more ways than the modifications they made.